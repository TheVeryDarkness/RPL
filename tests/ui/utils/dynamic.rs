#[rpl::dynamic(
    primary_message = "Dynamic RPL pattern",
    help = "You can use `#[rpl::dynamic]` to create a customizable lint.",
    note = "This is a dynamic RPL pattern, which can be customized during runtime."
)]
fn f1() {
    //~^ERROR: Dynamic RPL pattern
    //~|HELP: You can use `#[rpl::dynamic]` to create a customizable lint.
    //~|NOTE: This is a dynamic RPL pattern, which can be customized during runtime.
    //~|NOTE: `#[deny(rpl::dynamic)]` on by default
}

#[rpl::dynamic(
    primary_message = "Dynamic RPL pattern",
    unknown_attribute = "what's this?" 
    //~^ERROR: Unknown attribute key
    //~|NOTE: Allowed attribute keys are: `primary_message`, `labels`, `note`, `help`
)]
fn f2() {}

#[rpl::dynamic(help = "HELP!")] //~ERROR: Missing primary message
fn f3() {}

#[rpl::dynamic(primary_message = "test", help = 1)] //~ERROR: Expected a string value
fn f4() {}
