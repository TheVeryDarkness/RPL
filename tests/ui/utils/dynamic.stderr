error: Dynamic RPL pattern
  --> tests/ui/utils/dynamic.rs:6:1
   |
LL | fn f1() {
   | ^^^^^^^
   |
   = help: You can use `#[rpl::dynamic]` to create a customizable lint.
   = note: This is a dynamic RPL pattern, which can be customized during runtime.
   = note: `#[deny(rpl::dynamic)]` on by default

error: Unknown attribute key
  --> tests/ui/utils/dynamic.rs:15:5
   |
LL |     unknown_attribute = "what's this?" 
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: Allowed attribute keys are: `primary_message`, `labels`, `note`, `help`

error: Missing primary message
  --> tests/ui/utils/dynamic.rs:21:1
   |
LL | #[rpl::dynamic(help = "HELP!")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Expected a string value
  --> tests/ui/utils/dynamic.rs:24:42
   |
LL | #[rpl::dynamic(primary_message = "test", help = 1)]
   |                                          ^^^^^^^^

error: aborting due to 4 previous errors

